webui/src/main.tsx

```diff
@@ -66,17 +70,41 @@ class MainComponent extends React.Component<MainComponent.Props, MainComponent.S
             userLoading: true,
             error: '',
             isErrorDialogOpen: false,
-            isFooterExpanded: false
+            isFooterExpanded: false,
+            isBannerOpen: false
         };
     }
 
+    protected onDismissBannerButtonClick = () => {
+        const onClose = this.props.pageSettings.elements.banner?.props.onClose;
+        if (onClose) {
+            onClose();
+        }
+        const cookie = this.props.pageSettings.elements.banner?.props.cookieOnClose;
+        if (cookie) {
+            document.cookie = `${cookie.key}=${cookie.value}`;
+        }
+        this.setState({ isBannerOpen: false });
+    };
+
     componentDidMount(): void {
         const searchParams = new URLSearchParams(window.location.search);
         if (searchParams.has('auth-error')) {
             this.props.service.getUserAuthError()
                 .then(err => this.handleError(err));
         }
         this.updateUser();
+        const cookie = this.props.pageSettings.elements.banner?.props.cookieOnClose;
+        let open = true;
+        if (cookie) {
+            const decodedCookieString = decodeURIComponent(document.cookie);
```

Comment:
Is it valid / necessary to decode the whole `cookie` list upfront? From https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie I understand you can encode / decode individual values.